src/main/java/com/adobe/cq/testing/client/CQClient.java

```diff
@@ -205,6 +205,68 @@ public Boolean call() throws ClientException {
         return deletePagePolling.response;
     }
 
+    /**
+     * Returns whether a CQ page exists or not
+     * @param pagePath The path of the page
+     * @throws ClientException If the request failed
+     * @return whether the CQ page exists
+     */
+    public boolean pageExists(String pagePath) throws ClientException {
+        SlingHttpResponse response = getAuthorSitesPage(pagePath);
+        final int status = response.getStatusLine().getStatusCode();
+            return status == SC_OK;
+    }
+
+    /**
+     * Polls on whether a CQ page exists or not
+     * @param pagePath The path of the page
+     * @param timeout Timeout in milliseconds for the poller
+     * @throws InterruptedException if interrupted
+     * @return whether the CQ page exists
+     */
+    public boolean pageExistsWithRetry(String pagePath, int timeout) throws InterruptedException {
+        try {
+            new Polling(() -> pageExists(pagePath)).poll(timeout, 500);
+        } catch (TimeoutException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Polls on whether a CQ page exists or not after 1 second
+     * @param pagePath The path of the page
+     * @throws InterruptedException if interrupted
+     * @return whether the CQ page exists
+     */
+    public boolean pageExistsWithRetry(String pagePath) throws InterruptedException {
+        try {
+            new Polling(() -> pageExists(pagePath)).poll(1000, 500);
+        } catch (TimeoutException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Get a CQ Page under {code}/sites.html/{code} on author
+     * @param pagePath The path of the page
+     * @param expectedStatus An array of expected HTTP status codes for the response
+     * @return the http response
+     * @throws ClientException if the request failed
+     */
+    public SlingHttpResponse getAuthorSitesPage(String pagePath, int... expectedStatus) throws ClientException {
```

Comment:
FWIU, this method's goal is now to append `.html` and perform the request. If yes, the description and function name could be adapted accordingly.

Reply:
yup